crates/sui-core/src/epoch/reconfiguration.rs

```diff
@@ -192,29 +192,26 @@ where
         net_config.http2_keepalive_interval = Some(Duration::from_secs(5));
 
         for validator in next_epoch_validators {
-            let net_addr: &[u8] = &validator.net_address.clone();
-            let str_addr =
-                std::str::from_utf8(net_addr).map_err(|e| SuiError::GenericAuthorityError {
+            let address = Multiaddr::try_from(validator.net_address).map_err(|e| {
+                SuiError::GenericAuthorityError {
                     error: e.to_string(),
-                });
-            let address: Multiaddr = str_addr
-                .unwrap()
-                .parse()
-                .map_err(|e: multiaddr::Error| SuiError::GenericAuthorityError {
-                    error: e.to_string(),
-                })
-                .unwrap();
+                }
+            })?;
```

Comment:
Please create an issue and add a TODO here so we can get back to this: one invalid validator should not block the entire reconfiguration process. However what to do with the invalid validator in the new committee is an open question.

Reply:
Yeah, this is a good question. I think the best place to check this is in the move code. We ideally handle this here as well, but it would be more of a business / protocol decision what to do.